react/components/SKUSelector/index.tsx

```diff
@@ -199,20 +223,24 @@ const SKUSelectorContainer: FC<Props> = ({
     )
   }
 
-  const [
-    selectedVariations,
-    setSelectedVariations,
-  ] = useState<SelectedVariations>(() => getNewSelectedVariations(query, skuSelected, variations, initialSelection))
+  const [selectedVariations, setSelectedVariations] = useState<
+    SelectedVariations
+  >(() =>
+    getNewSelectedVariations(query, skuSelected, variations, initialSelection)
+  )
   useAllSelectedEvent(selectedVariations, variationsCount)
 
   useEffectSkipMount(() => {
-    setSelectedVariations(getNewSelectedVariations(query, skuSelected, variations, initialSelection))
+    setSelectedVariations(
+      getNewSelectedVariations(query, skuSelected, variations, initialSelection)
+    )
   }, [variations])
 
   useEffect(() => {
     if (skuSelected && onSKUSelected) {
       onSKUSelected(skuSelected.itemId)
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think it is better to remove this comment even if it will result in a warning

Reply:
And maybe add a comment explaining why it is ok to do what you are doing even though it results in a warning

Reply:
I agree with explaining why you are doing this, but keep the eslint-disable, so auto-formatters won't change it back